diff -NurB g/CMakeLists.txt f/CMakeLists.txt
--- g/CMakeLists.txt	2018-02-20 17:16:54.000000000 +0100
+++ f/CMakeLists.txt	2018-12-09 20:48:21.331005974 +0100
@@ -10,8 +10,8 @@
 
 PROJECT(BayesianCollaborativeDenoising)
 
-OPTION(BCD_BUILD_GUI "Build a GUI executable for bcd, using library NanoGUI" ON)
-OPTION(BCD_USE_CUDA "Use CUDA for accelerating computations" ON)
+OPTION(BCD_BUILD_GUI "Build a GUI executable for bcd, using library NanoGUI" OFF)
+OPTION(BCD_USE_CUDA "Use CUDA for accelerating computations" OFF)
 
 # Add CMake folder to the  module path : contains CMake specific files *.cmake
 SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/CMake" ${CMAKE_MODULE_PATH} )
@@ -31,32 +31,41 @@
 
 
 # OpenMP
-FIND_PACKAGE(OpenMP)
-
+FIND_PACKAGE(OpenMP REQUIRED)
+    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${OpenMP_C_FLAGS}")
+    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${OpenMP_CXX_FLAGS}")
+    set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_EXE_LINKER_FLAGS}")
+
+# Eigen
+find_package(Eigen3 REQUIRED)
+include_directories(${EIGEN3_INCLUDE_DIR})
+#OpenEXr
+find_package(OpenEXR REQUIRED)
+include_directories(${OPENXER_INCLUDE_DIR})
 #IF(${CMAKE_VERSION} VERSION_GREATER 3.8)
-#
-#	# not tested yet
-#	MESSAGE(AUTHOR_WARNING "Using OpenMP detection from cmake 3.9.0+: not tested yet")
-#	IF(OPENMP_CXX_FOUND)
-#		ADD_LIBRARY(OpenMP ALIAS OpenMP::OpenMP_CXX)
-#	ELSE()
-#		MESSAGE(WARNING "Could not find OpenMP library")
-#	ENDIF()
-#
-#ELSE()
 
-	IF(OPENMP_FOUND)
-		ADD_LIBRARY(OpenMP INTERFACE)
-		TARGET_COMPILE_OPTIONS(OpenMP INTERFACE ${OpenMP_CXX_FLAGS})
-		TARGET_LINK_LIBRARIES(OpenMP INTERFACE
-				$<$<CXX_COMPILER_ID:GNU>:${OpenMP_CXX_FLAGS}>
-				$<$<CXX_COMPILER_ID:Clang>:${OpenMP_CXX_FLAGS}>
-				$<$<CXX_COMPILER_ID:Intel>:${OpenMP_CXX_FLAGS}>
-				)
+	# not tested yet
+	MESSAGE(AUTHOR_WARNING "Using OpenMP detection from cmake 3.9.0+: not tested yet")
+	IF(OPENMP_CXX_FOUND)
+#		ADD_LIBRARY(OpenMP ALIAS OpenMP::OpenMP_CXX)
 	ELSE()
 		MESSAGE(WARNING "Could not find OpenMP library")
 	ENDIF()
 
+#ELSE()
+
+#	IF(OPENMP_FOUND)
+#		ADD_LIBRARY(OpenMP INTERFACE)
+#		TARGET_COMPILE_OPTIONS(OpenMP INTERFACE ${OpenMP_CXX_FLAGS})
+#		TARGET_LINK_LIBRARIES(OpenMP INTERFACE
+#				$<$<CXX_COMPILER_ID:GNU>:${OpenMP_CXX_FLAGS}>
+#				$<$<CXX_COMPILER_ID:Clang>:${OpenMP_CXX_FLAGS}>
+#				$<$<CXX_COMPILER_ID:Intel>:${OpenMP_CXX_FLAGS}>
+#				)
+#	ELSE()
+#		MESSAGE(WARNING "Could not find OpenMP library")
+#	ENDIF()
+
 #ENDIF()
 
 
@@ -112,7 +121,7 @@
 
 
 #MESSAGE(STATUS "Entering subdirectory 'ext'")
-ADD_SUBDIRECTORY(ext)
+#ADD_SUBDIRECTORY(ext)
 
 #MESSAGE(STATUS "Entering subdirectory 'src'")
 ADD_SUBDIRECTORY(src)
diff -NurB g/src/CMakeLists.txt f/src/CMakeLists.txt
--- g/src/CMakeLists.txt	2018-02-20 17:16:54.000000000 +0100
+++ f/src/CMakeLists.txt	2018-12-10 02:14:46.304090548 +0100
@@ -23,13 +23,13 @@
 ADD_SUBDIRECTORY(core)
 
 #MESSAGE(STATUS "Entering subdirectory 'io'")
-ADD_SUBDIRECTORY(io)
+#ADD_SUBDIRECTORY(io)
 
 #MESSAGE(STATUS "Entering subdirectory 'cli'")
-ADD_SUBDIRECTORY(cli)
+#ADD_SUBDIRECTORY(cli)
 
 #MESSAGE(STATUS "Entering subdirectory 'raw_converter'")
-ADD_SUBDIRECTORY(raw_converter)
+#ADD_SUBDIRECTORY(raw_converter)
 
 IF(BCD_BUILD_GUI)
 #	MESSAGE(STATUS "Entering subdirectory 'gui'")
diff -NurB g/src/core/SamplesAccumulator.cpp f/src/core/SamplesAccumulator.cpp
--- g/src/core/SamplesAccumulator.cpp	2018-02-20 17:16:54.000000000 +0100
+++ f/src/core/SamplesAccumulator.cpp	2018-12-09 18:48:02.131015358 +0100
@@ -11,6 +11,7 @@
 
 
 #include <cassert>
+#include <cmath>
 
 using namespace std;
 
diff -NurB g/src/core/SpikeRemovalFilter.cpp f/src/core/SpikeRemovalFilter.cpp
--- g/src/core/SpikeRemovalFilter.cpp	2018-02-20 17:16:54.000000000 +0100
+++ f/src/core/SpikeRemovalFilter.cpp	2018-12-09 18:49:21.739017884 +0100
@@ -9,6 +9,7 @@
 #include "SpikeRemovalFilter.h"
 
 #include "DeepImage.h"
+#include <cmath>
 
 using std::vector;
 
diff -NurB g/src/io/exr/io_exr.cpp f/src/io/exr/io_exr.cpp
--- g/src/io/exr/io_exr.cpp	2018-02-20 17:16:54.000000000 +0100
+++ f/src/io/exr/io_exr.cpp	2018-12-10 02:06:58.391075705 +0100
@@ -24,12 +24,12 @@
 
 /*VERSION 02.08.13*/
 
-#include <ImfInputFile.h>
-#include <ImfOutputFile.h>
-#include <ImfRgbaFile.h>
-#include <ImfChannelList.h>
-#include <ImfFrameBuffer.h>
-#include <half.h>
+#include <OpenEXR/ImfInputFile.h>
+#include <OpenEXR/ImfOutputFile.h>
+#include <OpenEXR/ImfRgbaFile.h>
+#include <OpenEXR/ImfChannelList.h>
+#include <OpenEXR/ImfFrameBuffer.h>
+#include <OpenEXR/half.h>
 #include <stdlib.h>
 #include <stdio.h>
 
