diff -NurB c/src/luxrays/core/context.cpp d/src/luxrays/core/context.cpp
--- c/src/luxrays/core/context.cpp	2018-06-12 12:33:01.096979999 +0200
+++ d/src/luxrays/core/context.cpp	2018-06-12 12:40:15.796954360 +0200
@@ -164,7 +164,12 @@
 
 std::vector<IntersectionDevice *> Context::CreateIntersectionDevices(
 	std::vector<DeviceDescription *> &deviceDesc, const size_t indexOffset) {
-	assert (!started);
+	if(started)
+	{
+		std::stringstream ss;
+		ss << "ERROR: " << __PRETTY_FUNCTION__ << " requires started to be unset, but it is";
+		throw std::runtime_error(ss.str());
+	}
 
 	LR_LOG(this, "Creating " << deviceDesc.size() << " intersection device(s)");
 
@@ -196,7 +201,12 @@
 }
 
 std::vector<IntersectionDevice *> Context::AddIntersectionDevices(std::vector<DeviceDescription *> &deviceDesc) {
-	assert (!started);
+	if(started)
+	{
+		std::stringstream ss;
+		ss << "ERROR: " << __PRETTY_FUNCTION__ << " requires started to be unset, but it is";
+		throw std::runtime_error(ss.str());
+	}
 
 	std::vector<IntersectionDevice *> newDevices = CreateIntersectionDevices(deviceDesc, idevices.size());
 	for (size_t i = 0; i < newDevices.size(); ++i)
@@ -207,7 +217,12 @@
 
 std::vector<IntersectionDevice *> Context::AddVirtualIntersectionDevice(
 	std::vector<DeviceDescription *> &deviceDescs) {
-	assert (!started);
+	if(started)
+	{
+		std::stringstream ss;
+		ss << "ERROR: " << __PRETTY_FUNCTION__ << " requires started to be unset, but it is";
+		throw std::runtime_error(ss.str());
+	}
 	assert (deviceDescs.size() > 0);
 
 	std::vector<IntersectionDevice *> realDevices = CreateIntersectionDevices(deviceDescs, idevices.size());
