diff -NurB c/src/luxrays/idevices/nativeidevice.cpp d/src/luxrays/idevices/nativeidevice.cpp
--- c/src/luxrays/idevices/nativeidevice.cpp	2018-06-12 12:33:01.096979999 +0200
+++ d/src/luxrays/idevices/nativeidevice.cpp	2018-06-18 16:41:12.418096071 +0200
@@ -130,7 +130,12 @@
 }
 
 RayBuffer *NativeThreadIntersectionDevice::PopRayBuffer(const u_int queueIndex) {
-	assert (started);
+	if(!started)
+	{
+		std::stringstream ss;
+		ss << "ERROR: " << __PRETTY_FUNCTION__ << " requires started to be set, but it isn't";
+		throw std::runtime_error(ss.str());
+	}
 	assert (dataParallelSupport);
 
 	return rayBufferQueue->PopDone(queueIndex);
diff -NurB c/src/luxrays/idevices/openclidevice.cpp d/src/luxrays/idevices/openclidevice.cpp
--- c/src/luxrays/idevices/openclidevice.cpp	2018-06-12 12:33:01.096979999 +0200
+++ d/src/luxrays/idevices/openclidevice.cpp	2018-06-18 16:41:23.174095437 +0200
@@ -221,7 +221,13 @@
 }
 
 RayBuffer *OpenCLIntersectionDevice::PopRayBuffer(const u_int queueIndex) {
-	assert (started);
+	if(!started)
+	{
+		std::stringstream ss;
+		ss << "ERROR: " << __PRETTY_FUNCTION__ << " requires started to be set, but it isn't";
+		throw std::runtime_error(ss.str());
+	}
+	
 	assert (dataParallelSupport);
 
 	return oclQueues[queueIndex]->PopRayBuffer();
