diff -NurB a/include/luxrays/utils/serializationutils.h LuxCore-luxcorerender_v2.2/include/luxrays/utils/serializationutils.h
--- a/include/luxrays/utils/serializationutils.h	2019-11-18 12:05:55.172002763 +0100
+++ LuxCore-luxcorerender_v2.2/include/luxrays/utils/serializationutils.h	2019-11-18 12:07:21.055003715 +0100
@@ -42,19 +42,19 @@
 #include <boost/serialization/set.hpp>
 #include <boost/serialization/vector.hpp>
 
-//#include <boost/archive/binary_iarchive.hpp>
-//#include <boost/archive/binary_oarchive.hpp>
-#include "eos/portable_oarchive.hpp"
-#include "eos/portable_iarchive.hpp"
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+//#include "eos/portable_oarchive.hpp"
+//#include "eos/portable_iarchive.hpp"
 
 #include "luxrays/luxrays.h"
 
 namespace luxrays {
 
-//typedef boost::archive::binary_oarchive LuxOutputArchive;
-//typedef boost::archive::binary_iarchive LuxInputArchive;
-typedef eos::portable_oarchive LuxOutputArchive;
-typedef eos::portable_iarchive LuxInputArchive;
+typedef boost::archive::binary_oarchive LuxOutputArchive;
+typedef boost::archive::binary_iarchive LuxInputArchive;
+//typedef eos::portable_oarchive LuxOutputArchive;
+//typedef eos::portable_iarchive LuxInputArchive;
 
 class SerializationOutputFile {
 public:
diff -NurB a/src/luxrays/utils/properties.cpp LuxCore-luxcorerender_v2.2/src/luxrays/utils/properties.cpp
--- a/src/luxrays/utils/properties.cpp	2019-11-18 12:05:55.244002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/luxrays/utils/properties.cpp	2019-11-18 12:14:10.205008253 +0100
@@ -30,6 +30,7 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/algorithm/string/trim.hpp>
 #include <boost/regex.hpp>
+#include <boost/lexical_cast.hpp>
 #include <boost/archive/iterators/base64_from_binary.hpp>
 #include <boost/archive/iterators/binary_from_base64.hpp>
 #include <boost/archive/iterators/transform_width.hpp>
diff -NurB a/src/slg/engines/bidircpu/bidircpurenderstate.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/bidircpu/bidircpurenderstate.cpp
--- a/src/slg/engines/bidircpu/bidircpurenderstate.cpp	2019-11-18 12:05:55.246002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/bidircpu/bidircpurenderstate.cpp	2019-11-18 13:32:09.938060149 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "luxrays/utils/serializationutils.h"
 #include "slg/engines/bidircpu/bidircpurenderstate.h"
 #include "slg/engines/bidircpu/bidircpu.h"
diff -NurB a/src/slg/engines/caches/photongi/pgicserialize.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/caches/photongi/pgicserialize.cpp
--- a/src/slg/engines/caches/photongi/pgicserialize.cpp	2019-11-18 12:05:55.246002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/caches/photongi/pgicserialize.cpp	2019-11-18 13:37:00.997063377 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "luxrays/utils/safesave.h"
 
 #include "slg/engines/caches/photongi/photongicache.h"
diff -NurB a/src/slg/engines/lightcpu/lightcpurenderstate.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/lightcpu/lightcpurenderstate.cpp
--- a/src/slg/engines/lightcpu/lightcpurenderstate.cpp	2019-11-18 12:05:55.246002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/lightcpu/lightcpurenderstate.cpp	2019-11-18 13:47:14.922070185 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "luxrays/utils/serializationutils.h"
 #include "slg/engines/lightcpu/lightcpurenderstate.h"
 #include "slg/engines/lightcpu/lightcpu.h"
diff -NurB a/src/slg/engines/pathcpu/pathcpurenderstate.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/pathcpu/pathcpurenderstate.cpp
--- a/src/slg/engines/pathcpu/pathcpurenderstate.cpp	2019-11-18 12:05:55.247002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/pathcpu/pathcpurenderstate.cpp	2019-11-18 13:43:09.732067466 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/engines/pathcpu/pathcpurenderstate.h"
 #include "slg/engines/pathcpu/pathcpu.h"
 #include "slg/engines/caches/photongi/photongicache.h"
diff -NurB a/src/slg/engines/pathocl/pathoclrenderstate.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/pathocl/pathoclrenderstate.cpp
--- a/src/slg/engines/pathocl/pathoclrenderstate.cpp	2019-11-18 12:05:55.246002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/pathocl/pathoclrenderstate.cpp	2019-11-18 13:40:49.021065905 +0100
@@ -18,6 +18,9 @@
 
 #if !defined(LUXRAYS_DISABLE_OPENCL)
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/engines/pathocl/pathoclrenderstate.h"
 #include "slg/engines/pathocl/pathocl.h"
 #include "slg/engines/caches/photongi/photongicache.h"
diff -NurB a/src/slg/engines/tilepathcpu/tilepathcpurenderstate.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/tilepathcpu/tilepathcpurenderstate.cpp
--- a/src/slg/engines/tilepathcpu/tilepathcpurenderstate.cpp	2019-11-18 12:05:55.246002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/tilepathcpu/tilepathcpurenderstate.cpp	2019-11-18 13:58:06.842077414 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/engines/tilerepository.h"
 #include "slg/engines/tilepathcpu/tilepathcpurenderstate.h"
 #include "slg/engines/tilepathcpu/tilepathcpu.h"
diff -NurB a/src/slg/engines/tilepathocl/tilepathoclrenderstate.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/tilepathocl/tilepathoclrenderstate.cpp
--- a/src/slg/engines/tilepathocl/tilepathoclrenderstate.cpp	2019-11-18 12:05:55.247002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/tilepathocl/tilepathoclrenderstate.cpp	2019-11-18 13:53:13.783074164 +0100
@@ -17,6 +17,8 @@
  ***************************************************************************/
 
 #if !defined(LUXRAYS_DISABLE_OPENCL)
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
 
 #include "slg/engines/tilerepository.h"
 #include "slg/engines/tilepathocl/tilepathoclrenderstate.h"
diff -NurB a/src/slg/engines/tilerepositoryserialize.cpp LuxCore-luxcorerender_v2.2/src/slg/engines/tilerepositoryserialize.cpp
--- a/src/slg/engines/tilerepositoryserialize.cpp	2019-11-18 12:05:55.247002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/engines/tilerepositoryserialize.cpp	2019-11-18 14:33:17.963100826 +0100
@@ -17,6 +17,8 @@
  ***************************************************************************/
 
 #include <boost/format.hpp>
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
 
 #include "slg/engines/tilerepository.h"
 
diff -NurB a/src/slg/film/convtest/filmconvtest.cpp LuxCore-luxcorerender_v2.2/src/slg/film/convtest/filmconvtest.cpp
--- a/src/slg/film/convtest/filmconvtest.cpp	2019-11-18 12:05:55.248002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/film/convtest/filmconvtest.cpp	2019-11-18 13:14:25.770048348 +0100
@@ -18,6 +18,9 @@
 
 #include <limits>
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/film/film.h"
 #include "slg/film/convtest/filmconvtest.h"
 #include "slg/film/imagepipeline/plugins/gaussianblur3x3.h"
diff -NurB a/src/slg/film/denoiser/samplesaccumulator.cpp LuxCore-luxcorerender_v2.2/src/slg/film/denoiser/samplesaccumulator.cpp
--- a/src/slg/film/denoiser/samplesaccumulator.cpp	2019-11-18 12:05:55.247002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/film/denoiser/samplesaccumulator.cpp	2019-11-18 13:16:51.246049961 +0100
@@ -30,6 +30,9 @@
 
 #include <bcd/core/CovarianceMatrix.h>
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "luxrays/utils/atomic.h"
 #include "slg/film/denoiser/samplesaccumulator.h"
 
diff -NurB a/src/slg/film/filmserialize.cpp LuxCore-luxcorerender_v2.2/src/slg/film/filmserialize.cpp
--- a/src/slg/film/filmserialize.cpp	2019-11-18 12:05:55.248002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/film/filmserialize.cpp	2019-11-18 13:04:35.888041806 +0100
@@ -20,7 +20,8 @@
 
 #include <boost/lexical_cast.hpp>
 #include <boost/foreach.hpp>
-
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
 #include "slg/film/film.h"
 
 using namespace std;
diff -NurB a/src/slg/film/imagepipeline/imagepipelineserialization.cpp LuxCore-luxcorerender_v2.2/src/slg/film/imagepipeline/imagepipelineserialization.cpp
--- a/src/slg/film/imagepipeline/imagepipelineserialization.cpp	2019-11-18 12:05:55.248002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/film/imagepipeline/imagepipelineserialization.cpp	2019-11-18 14:30:57.483099268 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/film/imagepipeline/imagepipeline.h"
 
 using namespace std;
diff -NurB a/src/slg/film/imagepipeline/radiancechannelscaleserialization.cpp LuxCore-luxcorerender_v2.2/src/slg/film/imagepipeline/radiancechannelscaleserialization.cpp
--- a/src/slg/film/imagepipeline/radiancechannelscaleserialization.cpp	2019-11-18 12:05:55.248002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/film/imagepipeline/radiancechannelscaleserialization.cpp	2019-11-18 13:21:09.749052828 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/film/imagepipeline/radiancechannelscale.h"
 
 using namespace std;
diff -NurB a/src/slg/film/noiseestimation/filmnoiseestimation.cpp LuxCore-luxcorerender_v2.2/src/slg/film/noiseestimation/filmnoiseestimation.cpp
--- a/src/slg/film/noiseestimation/filmnoiseestimation.cpp	2019-11-18 12:05:55.248002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/film/noiseestimation/filmnoiseestimation.cpp	2019-11-18 13:14:10.530048179 +0100
@@ -18,6 +18,9 @@
 
 #include <limits>
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/film/film.h"
 #include "slg/film/noiseestimation/filmnoiseestimation.h"
 
diff -NurB a/src/slg/imagemap/imagemapcacheserialize.cpp LuxCore-luxcorerender_v2.2/src/slg/imagemap/imagemapcacheserialize.cpp
--- a/src/slg/imagemap/imagemapcacheserialize.cpp	2019-11-18 12:05:55.249002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/imagemap/imagemapcacheserialize.cpp	2019-11-18 13:17:27.744050366 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/imagemap/imagemapcache.h"
 #include "slg/core/sdl.h"
 
diff -NurB a/src/slg/imagemap/imagemapserialize.cpp LuxCore-luxcorerender_v2.2/src/slg/imagemap/imagemapserialize.cpp
--- a/src/slg/imagemap/imagemapserialize.cpp	2019-11-18 12:05:55.249002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/imagemap/imagemapserialize.cpp	2019-11-18 14:09:59.958085323 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/imagemap/imagemap.h"
 
 using namespace std;
diff -NurB a/src/slg/renderconfig.cpp LuxCore-luxcorerender_v2.2/src/slg/renderconfig.cpp
--- a/src/slg/renderconfig.cpp	2019-11-18 12:05:55.251002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/renderconfig.cpp	2019-11-18 14:31:46.711099814 +0100
@@ -23,6 +23,8 @@
 #include <boost/algorithm/string.hpp> 
 #include <boost/filesystem.hpp>
 #include <boost/regex.hpp>
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
 
 #include "luxrays/utils/serializationutils.h"
 #include "slg/renderconfig.h"
diff -NurB a/src/slg/scene/extmeshcacheserialize.cpp LuxCore-luxcorerender_v2.2/src/slg/scene/extmeshcacheserialize.cpp
--- a/src/slg/scene/extmeshcacheserialize.cpp	2019-11-18 12:05:55.251002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/scene/extmeshcacheserialize.cpp	2019-11-18 14:29:06.700098039 +0100
@@ -16,6 +16,9 @@
  * limitations under the License.                                          *
  ***************************************************************************/
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "slg/scene/extmeshcache.h"
 
 using namespace std;
diff -NurB a/src/slg/scene/sceneserialize.cpp LuxCore-luxcorerender_v2.2/src/slg/scene/sceneserialize.cpp
--- a/src/slg/scene/sceneserialize.cpp	2019-11-18 12:05:55.251002764 +0100
+++ LuxCore-luxcorerender_v2.2/src/slg/scene/sceneserialize.cpp	2019-11-18 14:30:32.417098990 +0100
@@ -21,6 +21,9 @@
 #include <boost/lexical_cast.hpp>
 #include <boost/foreach.hpp>
 
+#include <boost/archive/polymorphic_iarchive.hpp>
+#include <boost/archive/polymorphic_oarchive.hpp>
+
 #include "luxrays/utils/serializationutils.h"
 #include "slg/scene/scene.h"
 
