diff -NurB a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
--- a/cmake/Dependencies.cmake	2018-11-13 09:44:09.000000000 +0100
+++ b/cmake/Dependencies.cmake	2018-12-02 22:13:44.097362349 +0100
@@ -127,6 +127,15 @@
 	include_directories(BEFORE SYSTEM ${BLOSC_INCLUDE_PATH})
 endif ()
 
+# OpenVDB
+set(OPENVDB_ROOT                   "${OPENVDB_SEARCH_PATH}")
+
+find_package(OpenVDB REQUIRED)
+
+if (OpenVDB_FOUND)
+	include_directories(BEFORE SYSTEM ${OPENVDB_INCLUDE_PATH})
+endif ()
+
 # OpenMP
 if(NOT APPLE)
 	find_package(OpenMP)
diff -NurB a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2018-11-13 09:44:09.000000000 +0100
+++ b/CMakeLists.txt	2018-12-02 22:16:03.637366776 +0100
@@ -110,6 +110,11 @@
 	SET(LUXRAYS_DISABLE_OPENCL 1)
 endif()
 
+if (NOT OpenVDB_FOUND)
+	MESSAGE(FATAL_ERROR "--> Could not locate required OpenVDB files - Please check ${OPENVDB_SEARCH_PATH}")
+endif()
+
+
 if (NOT OPENGL_FOUND AND NOT LUXRAYS_DISABLE_OPENCL)
 	MESSAGE(ERROR "--> Could not locate required OpenGL files, disabling OpenCL support, disabling samples build")
 	SET(LUXRAYS_DISABLE_OPENCL 1)
@@ -141,7 +146,7 @@
 	set(LUXCORE_LIBRARY luxcore)
 	ADD_DEFINITIONS("-DLUXCORE_DLL")
 else()
-	set(LUXCORE_LIBRARY luxcore slg-core slg-film slg-kernels luxrays bcd openvdb ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	set(LUXCORE_LIBRARY luxcore slg-core slg-film slg-kernels luxrays bcd ${OpenVDB_LIBRARIES} ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 endif()
 
 ################################################################################
diff -NurB a/samples/luxcorescenedemo/CMakeLists.txt b/samples/luxcorescenedemo/CMakeLists.txt
--- a/samples/luxcorescenedemo/CMakeLists.txt	2018-11-13 09:44:09.000000000 +0100
+++ b/samples/luxcorescenedemo/CMakeLists.txt	2018-12-02 22:22:51.012379699 +0100
@@ -29,4 +29,4 @@
 add_executable(luxcorescenedemo ${LUXCORESCENEDEMO_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 
-TARGET_LINK_LIBRARIES(luxcorescenedemo ${LUXCORE_LIBRARY} ${Boost_LIBRARIES} ${OPENCL_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcorescenedemo ${LUXCORE_LIBRARY} ${OpenVDB_LIBRARIES} ${Boost_LIBRARIES} ${OPENCL_LIBRARIES})
diff -NurB a/src/luxcore/CMakeLists.txt b/src/luxcore/CMakeLists.txt
--- a/src/luxcore/CMakeLists.txt	2018-11-13 09:44:09.000000000 +0100
+++ b/src/luxcore/CMakeLists.txt	2018-12-02 22:25:11.596384159 +0100
@@ -150,7 +150,7 @@
 
 if (BUILD_LUXCORE_DLL)
 	add_library(luxcore SHARED ${LUXCORE_LIB_SRCS} ${LUX_PARSER_SRC})
-	target_link_libraries(luxcore PRIVATE slg-core slg-film slg-kernels luxrays bcd openvdb ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	target_link_libraries(luxcore PRIVATE slg-core slg-film slg-kernels luxrays bcd ${OpenVDB_LIBRARIES} ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 else()
 	add_library(luxcore STATIC ${LUXCORE_LIB_SRCS} ${LUX_PARSER_SRC})
 endif()
@@ -174,10 +174,10 @@
 add_library(pyluxcore MODULE ${PYLUXCORE_SRCS} ${LUXCORE_LIB_SRCS} ${LUX_PARSER_SRC})
 add_definitions(${VISIBILITY_FLAGS})
 if(APPLE)
-	target_link_libraries(pyluxcore -Wl,-undefined -Wl,dynamic_lookup slg-core slg-film slg-kernels luxrays bcd openvdb ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	target_link_libraries(pyluxcore -Wl,-undefined -Wl,dynamic_lookup slg-core slg-film slg-kernels luxrays bcd ${OpenVDB_LIBRARIES} ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
     SET_TARGET_PROPERTIES(pyluxcore PROPERTIES XCODE_ATTRIBUTE_DEPLOYMENT_POSTPROCESSING NO) # exclude pylux from strip, not possible
 else()
-	target_link_libraries(pyluxcore PRIVATE slg-core slg-film slg-kernels luxrays bcd openvdb ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	target_link_libraries(pyluxcore PRIVATE slg-core slg-film slg-kernels luxrays bcd ${OpenVDB_LIBRARIES} ${BLOSC_LIBRARIES} ${EMBREE_LIBRARY} ${TBB_LIBRARY} ${TIFF_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 endif()
 
 set_target_properties(pyluxcore PROPERTIES PREFIX "")
