diff -NurB a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
--- a/cmake/Dependencies.cmake	2022-08-08 22:41:10.636894634 +0200
+++ b/cmake/Dependencies.cmake	2022-08-08 22:48:49.113865873 +0200
@@ -43,11 +43,7 @@
     find_package(PNG REQUIRED)
     include_directories(BEFORE SYSTEM ${PNG_PNG_INCLUDE_DIR})
 	# Find Python Libraries
-    if("${PYTHON_V}" EQUAL "27")
-        find_package(PythonLibs 2.7)
-    else()
-        find_package(PythonLibs 3.4)
-    endif()
+	find_package (Python COMPONENTS Interpreter Development)
 endif()
 
 find_program(PYSIDE_UIC NAMES pyside-uic pyside2-uic pyside6-uic
