diff -NurB b/src/slg/CMakeLists.txt c/src/slg/CMakeLists.txt
--- b/src/slg/CMakeLists.txt	2018-11-13 09:44:09.000000000 +0100
+++ c/src/slg/CMakeLists.txt	2018-12-10 09:30:47.267511518 +0100
@@ -78,69 +78,6 @@
 
 add_definitions(-DCPP_API_EXPORTS ${VISIBILITY_FLAGS}) # for controlling visibility
 
-###########################################################################
-#
-# BCD files
-#
-###########################################################################
-
-set(BCD_SRCS
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/Chronometer.cpp
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/CovarianceMatrix.cpp
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/Denoiser.cpp
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/DenoisingUnit.cpp
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/MultiscaleDenoiser.cpp
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/SamplesAccumulator.cpp
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/SpikeRemovalFilter.cpp
-	${LuxRays_SOURCE_DIR}/deps/bcd-1.1/src/core/Utils.cpp
-)
-
-SOURCE_GROUP("Source Files\\BCD Library" FILES ${BCD_SRCS})
-
-include_directories(${LuxRays_SOURCE_DIR}/deps/eigen-3.3.4)
-include_directories(${LuxRays_SOURCE_DIR}/deps/bcd-1.1/include)
-include_directories(${LuxRays_SOURCE_DIR}/deps/bcd-1.1/include/bcd/core)
-
-# Use only MPL licensed source code
-add_definitions("-DEIGEN_MPL2_ONLY")
-
-add_library(bcd STATIC ${BCD_SRCS})
-
-###########################################################################
-#
-# OpenVDB files
-#
-###########################################################################
-
-set(OPENVDB_SRCS
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/Grid.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/Platform.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/openvdb.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/io/Archive.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/io/Compression.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/io/File.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/io/GridDescriptor.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/io/Queue.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/io/Stream.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/io/TempFile.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/math/Maps.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/math/Proximity.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/math/QuantizedUnitVec.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/math/Transform.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/metadata/MetaMap.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/metadata/Metadata.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/util/Formats.cc
-	${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0/openvdb/util/Util.cc
-)
-
-SOURCE_GROUP("Source Files\\OpenVDB Library" FILES ${OPENVDB_SRCS})
-
-include_directories(${LuxRays_SOURCE_DIR}/deps/openvdb-3.1.0)
-
-# Required by OpenVDB to read ABI 3
-add_definitions("-DOPENVDB_3_ABI_COMPATIBLE -DOPENVDB_STATICLIB -DOPENVDB_OPENEXR_STATICLIB")
-
-add_library(openvdb STATIC ${OPENVDB_SRCS})
 
 ###########################################################################
 #
