diff -NurB b/samples/benchsimple/CMakeLists.txt c/samples/benchsimple/CMakeLists.txt
--- b/samples/benchsimple/CMakeLists.txt	2022-10-02 23:59:14.831578971 +0200
+++ c/samples/benchsimple/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -21,4 +21,4 @@
 
 add_executable(benchsimple benchsimple.cpp)
 add_definitions(${VISIBILITY_FLAGS})
-target_link_libraries(benchsimple luxrays ${Embree_LIBRARY})
+target_link_libraries(benchsimple luxrays ${Embree_LIBRARY}  Python::Python OpenCL::OpenCL embree)
diff -NurB b/samples/luxcoreconsole/CMakeLists.txt c/samples/luxcoreconsole/CMakeLists.txt
--- b/samples/luxcoreconsole/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcoreconsole/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -28,4 +28,4 @@
 
 add_executable(luxcoreconsole ${LUXCORECONSOLE_SRCS})
 
-TARGET_LINK_LIBRARIES(luxcoreconsole luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoreconsole luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES}  Python::Python OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/luxcoredemo/CMakeLists.txt c/samples/luxcoredemo/CMakeLists.txt
--- b/samples/luxcoredemo/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcoredemo/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -29,4 +29,4 @@
 add_executable(luxcoredemo ${LUXCORELIBDEMO_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 
-TARGET_LINK_LIBRARIES(luxcoredemo luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoredemo luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES} Python::Python OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/luxcoreimplserializationdemo/CMakeLists.txt c/samples/luxcoreimplserializationdemo/CMakeLists.txt
--- b/samples/luxcoreimplserializationdemo/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcoreimplserializationdemo/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -28,4 +28,4 @@
 
 add_executable(luxcoreimplserializationdemo ${LUXCOREIMPL_SERIALIZATIONDEMO_SRCS})
 
-TARGET_LINK_LIBRARIES(luxcoreimplserializationdemo luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoreimplserializationdemo luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES} Python::Python OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/luxcorescenedemo/CMakeLists.txt c/samples/luxcorescenedemo/CMakeLists.txt
--- b/samples/luxcorescenedemo/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcorescenedemo/CMakeLists.txt	2022-10-03 00:27:56.226517680 +0200
@@ -29,4 +29,4 @@
 add_executable(luxcorescenedemo ${LUXCORESCENEDEMO_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 
-TARGET_LINK_LIBRARIES(luxcorescenedemo luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcorescenedemo luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES} Python::Python OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/luxcoreui/CMakeLists.txt c/samples/luxcoreui/CMakeLists.txt
--- b/samples/luxcoreui/CMakeLists.txt	2022-10-02 23:59:14.838578971 +0200
+++ c/samples/luxcoreui/CMakeLists.txt	2022-10-03 00:27:56.232517680 +0200
@@ -84,6 +84,8 @@
 	glfw
 	${GLFW_LIBRARIES}
 	luxcore smallluxgpu luxrays
+	 Python::Python OpenCL::OpenCL
+	OpenImageIO::OpenImageIO embree
 	${Embree_LIBRARY}
 	${OPENGL_LIBRARIES}
 	${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES}
diff -NurB b/samples/smallluxgpu4/CMakeLists.txt c/samples/smallluxgpu4/CMakeLists.txt
--- b/samples/smallluxgpu4/CMakeLists.txt	2022-10-02 23:59:14.838578971 +0200
+++ c/samples/smallluxgpu4/CMakeLists.txt	2022-10-03 00:27:56.233517680 +0200
@@ -32,7 +32,7 @@
 if(APPLE)
 	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 else(APPLE)
-	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES} OpenCL::OpenCL OpenImageIO::OpenImageIO Python::Python embree glut)
 endif(APPLE)
 
 # This instructs FREEGLUT to emit a pragma for the static version
