--- a/cmake/Dependencies.cmake	2022-10-17 13:41:31.479457391 +0200
+++ LuxCore-luxrender_v1.6/cmake/Dependencies.cmake	2022-10-17 14:11:09.057394099 +0200
@@ -16,8 +16,8 @@
 # limitations under the License.
 ################################################################################
 
-include(FindPkgMacros)
-getenv_path(LuxRays_DEPENDENCIES_DIR)
+#include(FindPkgMacros)
+#getenv_path(LuxRays_DEPENDENCIES_DIR)
 
 ################################################################################
 #
@@ -32,6 +32,43 @@
 include_directories(BEFORE SYSTEM ${OPENIMAGEIO_INCLUDE_DIR})
 find_package(OpenEXR REQUIRED)
 
+set(PythonVersions "" CACHE STRING "Eglible python versions") 
+
+function(TestPythonVersion TestVersion)
+	message(STATUS "Testing PythonVersion ${TestVersion}")
+	find_package(Python ${TestVersion} EXACT COMPONENTS Interpreter Development)
+	if(NOT ${Python_FOUND})
+		set(PythonVersionEgligible OFF PARENT_SCOPE)
+		return()
+	endif()
+# Find Boost
+set(Boost_USE_STATIC_LIBS       OFF)
+set(Boost_USE_MULTITHREADED     ON)
+set(Boost_USE_STATIC_RUNTIME    OFF)
+set(BOOST_ROOT                  "${BOOST_SEARCH_PATH}")
+#set(Boost_DEBUG                 ON)
+set(Boost_MINIMUM_VERSION       "1.44.0")
+
+
+set(LUXRAYS_BOOST_COMPONENTS thread program_options filesystem serialization iostreams regex system "python${Python_VERSION_MAJOR}${Python_VERSION_MINOR}" chrono)
+find_package(Boost ${Boost_MINIMUM_VERSION} COMPONENTS ${LUXRAYS_BOOST_COMPONENTS})
+if (NOT Boost_FOUND)
+        # Try again with the other type of libs
+        if(Boost_USE_STATIC_LIBS)
+                set(Boost_USE_STATIC_LIBS OFF)
+        else()
+                set(Boost_USE_STATIC_LIBS ON)
+        endif()
+        find_package(Boost ${Boost_MINIMUM_VERSION} COMPONENTS ${LUXRAYS_BOOST_COMPONENTS})
+endif()
+	if(NOT ${Boost_FOUND})
+		set(PythonVersionEligible OFF PARENT_SCOPE)
+		return()
+	endif()
+	set(PythonVersionEligible ON PARENT_SCOPE)
+
+endfunction()
+
 if(NOT APPLE)
     # Apple has these available hardcoded and matched in macos repo, see Config_OSX.cmake
 
@@ -42,11 +79,25 @@
     include_directories(BEFORE SYSTEM ${JPEG_INCLUDE_DIR})
     find_package(PNG REQUIRED)
     include_directories(BEFORE SYSTEM ${PNG_PNG_INCLUDE_DIR})
-	# Find Python Libraries
-	find_package(PythonLibs)
+
+	list(REVERSE PythonVersions)
+        message(STATUS "PythonsVersions: ${PythonVersions}")
+        foreach(PythonVersionCandidate IN LISTS PythonVersions )
+                if("X${PythonVersionCandidate}" STREQUAL "X")
+			continue()
+                endif()
+                message(STATUS "PythonsVersionCandidate: ${PythonVersionCandidate}")
+                TestPythonVersion("${PythonVersionCandidate}")
+                if(${PythonVersionEligible})
+			find_package(Python ${PythonVersionCandidate} EXACT COMPONENTS Interpreter Development)
+                        message(STATUS "Going with Python ${PythonVersionCandidate}")
+                        break()
+                endif()
+        endforeach()
 endif()
+#find_package(Python 3.7 EXACT COMPONENTS Interpreter Development)
 
-include_directories(${PYTHON_INCLUDE_DIRS})
+include_directories(${Python_INCLUDE_DIRS})
 
 # Find Boost
 set(Boost_USE_STATIC_LIBS       OFF)
@@ -56,9 +107,8 @@
 #set(Boost_DEBUG                 ON)
 set(Boost_MINIMUM_VERSION       "1.44.0")
 
-set(Boost_ADDITIONAL_VERSIONS "1.47.0" "1.46.1" "1.46" "1.46.0" "1.45" "1.45.0" "1.44" "1.44.0")
+set(LUXRAYS_BOOST_COMPONENTS thread program_options filesystem serialization iostreams regex system "python${Python_VERSION_MAJOR}${Python_VERSION_MINOR}" chrono)
 
-set(LUXRAYS_BOOST_COMPONENTS thread program_options filesystem serialization iostreams regex system python chrono)
 find_package(Boost ${Boost_MINIMUM_VERSION} COMPONENTS ${LUXRAYS_BOOST_COMPONENTS})
 if (NOT Boost_FOUND)
         # Try again with the other type of libs
@@ -71,6 +121,7 @@
 endif()
 
 if (Boost_FOUND)
+	message(STATUS "Boost found ${Boost_VERSION}")
 	include_directories(BEFORE SYSTEM ${Boost_INCLUDE_DIRS})
 	link_directories(${Boost_LIBRARY_DIRS})
 	# Don't use old boost versions interfaces
@@ -81,7 +132,7 @@
 	endif()
 endif ()
 
-
+if(BUILD_SAMPELS_OPENGL)
 # OpenGL
 find_package(OpenGL)
 
@@ -106,21 +157,24 @@
 if (GLUT_FOUND)
 	include_directories(BEFORE SYSTEM ${GLUT_INCLUDE_PATH})
 endif ()
+endif(BUILD_SAMPELS_OPENGL)
 
+if(NOT LUXRAYS_DISABLE_OPENCL)
 # OpenCL
-set(OPENCL_ROOT                "${OPENCL_SEARCH_PATH}")
 find_package(OpenCL)
 
-if (OPENCL_FOUND)
-	include_directories(BEFORE SYSTEM ${OPENCL_INCLUDE_DIR} ${OPENCL_C_INCLUDE_DIR})
+if (OpenCL_FOUND)
+	include_directories(BEFORE SYSTEM ${OpenCL_INCLUDE_DIR} ${OpenCL_C_INCLUDE_DIR})
 endif ()
+endif(NOT LUXRAYS_DISABLE_OPENCL)
 
 # Intel Embree
 set(EMBREE_ROOT                "${EMBREE_SEARCH_PATH}")
-find_package(Embree)
+find_package(Embree 3)
 
-if (EMBREE_FOUND)
-	include_directories(BEFORE SYSTEM ${EMBREE_INCLUDE_PATH})
+if (Embree_FOUND)
+	message(STATUS              "Embree found: ${Embree_VERSION}")
+	include_directories(BEFORE SYSTEM ${Embree_INCLUDE_PATH})
 endif ()
 
 # OpenMP
--- a/CMakeLists.txt	2022-10-17 13:41:31.478457391 +0200
+++ LuxCore-luxrender_v1.6/CMakeLists.txt	2022-10-17 14:23:36.355367491 +0200
@@ -25,11 +25,13 @@
 #
 ################################################################################
 # Fresh start
-cmake_minimum_required(VERSION 2.8)
-cmake_policy(VERSION 2.8)
-#Remove the following when the version check is at least 2.8.4
-SET(CMAKE_LEGACY_CYGWIN_WIN32 0)
+cmake_minimum_required(VERSION 3.20)
+cmake_policy(VERSION 3.20)
 
+include(CMakeDependentOption)
+
+option(BUILD_SAMPLES "Build samples")
+cmake_dependent_option(BUILD_SAMPLES_OPENGL "Build samples requiring OpenGL" OFF "BUILD_SAMPLES" OFF )
 project(LuxRays)
 
 # Don't over-configure
@@ -54,9 +56,9 @@
 # cmake does need to know which of the possible configurations
 # to generate
 if(NOT CMAKE_BUILD_TYPE)
-	set(CMAKE_BUILD_TYPE Release)
+#_cmake_modify_IGNORE 	set(CMAKE_BUILD_TYPE Release)
 endif(NOT CMAKE_BUILD_TYPE)
-SET(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE} CACHE STRING "assure config" FORCE) # makes sure type is shown in cmake gui
+#_cmake_modify_IGNORE SET(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE} CACHE STRING "assure config" FORCE) # makes sure type is shown in cmake gui
 
 message(STATUS "Building mode: " ${CMAKE_BUILD_TYPE})
 
@@ -64,8 +66,6 @@
 INCLUDE(Configuration)
 INCLUDE(KernelPreprocess)
 
-# Install CMake modules
-#add_subdirectory(CMake)
 
 SET(LuxRays_INCLUDE_DIR "${LuxRays_SOURCE_DIR}/include")
 include_directories("${LuxRays_INCLUDE_DIR}")
@@ -77,20 +77,15 @@
 	MESSAGE(FATAL_ERROR "--> Could not locate required Boost files - Please check ${BOOST_SEARCH_PATH}")
 endif()
 
-if (NOT OPENIMAGEIO_FOUND)
+if (NOT OpenImageIO_FOUND)
 	MESSAGE(FATAL_ERROR "--> Could not locate required OpenImageIO files - Please check ${OPENIMAGEIO_SEARCH_PATH}")
 endif()
 
-if (NOT OPENCL_FOUND AND NOT LUXRAYS_DISABLE_OPENCL)
-	MESSAGE(ERROR "--> Could not locate required OpenCL files, disabling OpenCL support - Please check ${OPENCL_SEARCH_PATH}")
-	SET(LUXRAYS_DISABLE_OPENCL 1)
-endif()
-
-if (NOT OPENGL_FOUND AND NOT LUXRAYS_DISABLE_OPENCL)
-	MESSAGE(ERROR "--> Could not locate required OpenGL files, disabling OpenCL support, disabling samples build")
-	SET(LUXRAYS_DISABLE_OPENCL 1)
+if (NOT OpenCL_FOUND AND NOT LUXRAYS_DISABLE_OPENCL)
+	MESSAGE(FATAL_ERROR "--> Could not locate required OpenCL files")
 endif()
 
+if(BUILD_SAMPLES_OPENGL)
 if (NOT GLEW_FOUND)
 	MESSAGE(FATAL_ERROR "--> Could not locate required GLEW files, disabling samples build - Please check ${GLEW_SEARCH_PATH}")
 endif()
@@ -98,9 +93,10 @@
 if (NOT GLUT_FOUND)
 	MESSAGE(FATAL_ERROR "--> Could not locate required GLUT files, disabling samples build - Please check ${GLUT_SEARCH_PATH}")
 endif()
+endif(BUILD_SAMPLES_OPENGL)
 
-if (NOT EMBREE_FOUND)
-	MESSAGE(FATAL_ERROR "--> Could not locate required Intel Embree files - Please check ${EMBREE_SEARCH_PATH}")
+if (NOT Embree_FOUND)
+	MESSAGE(FATAL_ERROR "--> Could not locate required Intel Embree files - Please check ${Embree_SEARCH_PATH}")
 endif()
 
 if (LUXRAYS_DISABLE_OPENCL)
@@ -128,22 +124,17 @@
 #
 ################################################################################
 
-if(NOT APPLE OR OSX_BUILD_DEMOS)
+if(BUILD_SAMPLES)
 	add_subdirectory(samples/benchsimple)
 	add_subdirectory(samples/luxcoredemo)
 	add_subdirectory(samples/luxcorescenedemo)
 	add_subdirectory(samples/luxcoreimplserializationdemo)
-endif()
-
-if(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
-	add_subdirectory(samples/smallluxgpu4)
-endif(OPENGL_FOUND AND GLUT_FOUND AND GLEW_FOUND)
-
-add_subdirectory(samples/luxcoreconsole)
-if(OPENGL_FOUND)
-	add_subdirectory(samples/luxcoreui)
-endif(OPENGL_FOUND)
-
+	add_subdirectory(samples/luxcoreconsole)
+	if(BUILD_SAMPLES_OPENGL)
+		add_subdirectory(samples/smallluxgpu4)
+		add_subdirectory(samples/luxcoreui)
+	endif(BUILD_SAMPLES_OPENGL)
+endif(BUILD_SAMPLES)
 
 ################################################################################
 #
diff -NurB b/samples/benchsimple/CMakeLists.txt c/samples/benchsimple/CMakeLists.txt
--- b/samples/benchsimple/CMakeLists.txt	2022-10-02 23:59:14.831578971 +0200
+++ c/samples/benchsimple/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -21,4 +21,4 @@
 
 add_executable(benchsimple benchsimple.cpp)
 add_definitions(${VISIBILITY_FLAGS})
-target_link_libraries(benchsimple luxrays ${EMBREE_LIBRARY})
+target_link_libraries(benchsimple luxrays OpenCL::OpenCL embree)
diff -NurB b/samples/luxcoreconsole/CMakeLists.txt c/samples/luxcoreconsole/CMakeLists.txt
--- b/samples/luxcoreconsole/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcoreconsole/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -28,4 +28,4 @@
 
 add_executable(luxcoreconsole ${LUXCORECONSOLE_SRCS})
 
-TARGET_LINK_LIBRARIES(luxcoreconsole luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoreconsole luxcore smallluxgpu luxrays TIFF::TIFF OpenEXR::OpenEXR PNG::PNG JPEG::JPEG OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/luxcoredemo/CMakeLists.txt c/samples/luxcoredemo/CMakeLists.txt
--- b/samples/luxcoredemo/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcoredemo/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -29,4 +29,4 @@
 add_executable(luxcoredemo ${LUXCORELIBDEMO_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 
-TARGET_LINK_LIBRARIES(luxcoredemo luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoredemo luxcore smallluxgpu luxrays TIFF::TIFF OpenEXR::OpenEXR PNG::PNG JPEG::JPEG OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/luxcoreimplserializationdemo/CMakeLists.txt c/samples/luxcoreimplserializationdemo/CMakeLists.txt
--- b/samples/luxcoreimplserializationdemo/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcoreimplserializationdemo/CMakeLists.txt	2022-10-03 00:27:56.225517680 +0200
@@ -28,4 +28,4 @@
 
 add_executable(luxcoreimplserializationdemo ${LUXCOREIMPL_SERIALIZATIONDEMO_SRCS})
 
-TARGET_LINK_LIBRARIES(luxcoreimplserializationdemo luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcoreimplserializationdemo luxcore smallluxgpu luxrays TIFF::TIFF OpenEXR::OpenEXR PNG::PNG JPEG::JPEG OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/luxcorescenedemo/CMakeLists.txt c/samples/luxcorescenedemo/CMakeLists.txt
--- b/samples/luxcorescenedemo/CMakeLists.txt	2022-10-02 23:59:14.832578971 +0200
+++ c/samples/luxcorescenedemo/CMakeLists.txt	2022-10-03 00:27:56.226517680 +0200
@@ -29,4 +29,4 @@
 add_executable(luxcorescenedemo ${LUXCORESCENEDEMO_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 
-TARGET_LINK_LIBRARIES(luxcorescenedemo luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+TARGET_LINK_LIBRARIES(luxcorescenedemo luxcore smallluxgpu luxrays TIFF::TIFF OpenEXR::OpenEXR PNG::PNG JPEG::JPEG OpenCL::OpenCL OpenImageIO::OpenImageIO embree)
diff -NurB b/samples/smallluxgpu4/CMakeLists.txt c/samples/smallluxgpu4/CMakeLists.txt
--- b/samples/smallluxgpu4/CMakeLists.txt	2022-10-02 23:59:14.838578971 +0200
+++ c/samples/smallluxgpu4/CMakeLists.txt	2022-10-03 00:27:56.233517680 +0200
@@ -32,7 +32,7 @@
 if(APPLE)
 	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${Embree_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
 else(APPLE)
-	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${OPENGL_LIBRARIES} ${GLEW_LIBRARY} ${GLUT_LIBRARY} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	TARGET_LINK_LIBRARIES(slg4 luxcore smallluxgpu luxrays glew glut TIFF::TIFF OpenEXR::OpenEXR PNG::PNG JPEG::JPEG OpenCL::OpenCL OpenImageIO::OpenImageIO embree glut)
 endif(APPLE)
 
 # This instructs FREEGLUT to emit a pragma for the static version
diff -Nurb LuxCore-luxrender_v1.6/src/luxcore/CMakeLists.txt c/src/luxcore/CMakeLists.txt
--- LuxCore-luxrender_v1.6/src/luxcore/CMakeLists.txt	2016-05-09 22:17:57.000000000 +0200
+++ c/src/luxcore/CMakeLists.txt	2020-11-12 19:52:42.772166106 +0100
@@ -141,10 +141,10 @@
 add_library(pyluxcore MODULE ${PYLUXCORE_SRCS})
 add_definitions(${VISIBILITY_FLAGS})
 if(APPLE)
-	target_link_libraries(pyluxcore -Wl,-undefined -Wl,dynamic_lookup luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${Boost_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	target_link_libraries(pyluxcore -Wl,-undefined -Wl,dynamic_lookup luxcore smallluxgpu luxrays embree TIFF::TIFF OpenEXR::OpenEXR PNG::PNG JPEG::JPEG)
     SET_TARGET_PROPERTIES(pyluxcore PROPERTIES XCODE_ATTRIBUTE_DEPLOYMENT_POSTPROCESSING NO) # exclude pylux from strip, not possible
 else()
-	target_link_libraries(pyluxcore luxcore smallluxgpu luxrays ${EMBREE_LIBRARY} ${Boost_LIBRARIES} ${TIFF_LIBRARIES} ${OPENEXR_LIBRARIES} ${PNG_LIBRARIES} ${JPEG_LIBRARIES})
+	target_link_libraries(pyluxcore luxcore smallluxgpu luxrays embree TIFF::TIFF OpenEXR::OpenEXR PNG::PNG JPEG::JPEG)
 endif()
 
 set_target_properties(pyluxcore PROPERTIES PREFIX "")
