diff -Nurb a/src/luxrays/utils/ocl.cpp b/src/luxrays/utils/ocl.cpp
--- a/src/luxrays/utils/ocl.cpp	2018-03-27 02:25:21.873308274 +0200
+++ b/src/luxrays/utils/ocl.cpp	2018-03-27 02:55:54.493249200 +0200
@@ -237,7 +237,7 @@
 //------------------------------------------------------------------------------
 
 static std::string SanitizeFileName(const std::string &name) {
-	std::string sanitizedName = name;
+	std::string sanitizedName = name.c_str();
 
 	boost::replace_all(sanitizedName, ":", "__");
 	boost::replace_all(sanitizedName, "/", "__");
